readme.md

```diff
@@ -1,104 +1,175 @@
-
-![Concept map](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/platform-platform/monorepo/master/concept_map.puml)
+### Features
+- Authentication verification (by providing `AuthCredentials`);
+- Handling `GET`, `DELETE`, `POST`, `PUT` HTTP methods;
+- Flexible request URL matching (using `ExactPathMatcher`, `RegExpPathMatcher`).
 
 <details>
-  <summary>Metrics Components</summary>
-
-#### CI integrations
-
-A CLI application that integrates with popular CI tools to collect software project metrics.
-
-#### Core
-
-A Dart package that provides a common classes to use within Metrics applications.
-
-#### Firebase
-
-A `Firebase` instance that provides the Firestore, Firebase Cloud Functions services and ability to deploy the application on Firebase Hosting. Also, provides an Analytics service used to gather and store the analytics data (this service is optional and may not be configured during deployment).
+  <summary>Usage example</summary>
+
+Consider this short example on how to use the API Mock Server.
+
+First, let's consider we have the following API client, and we need to test its `fetchBar` method:
+```dart
+import 'package:http/http.dart' as http;
+
+class TestClient {
+  final String apiUrl;
+
+  const TestClient(this.apiUrl);
+
+  Future<String> fetchBar() async {
+    final response = await http.get('$apiUrl/foo');
+
+    if (response.statusCode != 200) return null;
+
+    return response.body;
+  }
+}
+```
+
+After that, let's implement an Api Mock server to test the `TestClient`.
+```dart
+class MockServer extends ApiMockServer {
+  @override
+  List<RequestHandler> get handlers => [
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher('/foo'),
+          dispatcher: _fooHandler,
+        ),
+      ];
+
+  Future<void> _fooHandler(HttpRequest request) async {
+    request.response.write('bar');
+    
+    await request.response.flush();
+    await request.response.close();
+  }
+}
+```
+
+Now, let's use the created `MockServer` in tests:
+```dart
+void main() {
+  group("TestClient", () {
+    final mockServer = MockServer();
+    TestClient client;
+
+    setUpAll(() async {
+      await mockServer.start();
+      client = TestClient(mockServer.url);
+    });
+
+    test(
+      ".fetchBar() returns 'bar'",
+      () async {
+        const expectedResponse = 'bar';
+
+        final actualResponse = await client.fetchBar();
+
+        expect(actualResponse, equals(expectedResponse));
+      },
+    );
+  });
+}
+```
+</details>
 
-Firebase Analytics is optional and may not be configured during deployment.
+## :shield: Guardian
+[Guardian](guardian) is a tool for detecting and publishing flaky tests by analyzing JUnit XML files. The tool accepts the unit test reports and compares the actual and previous test results stored in a database. If the test is considered flaky, Guardian notifies the team using Slack and creates an issue in Jira.
```

Comment:
```suggestion
[Guardian](guardian) is a tool for detecting and notifying about flaky tests by analyzing JUnit XML files. This tool accepts the unit test reports and compares the actual and previous test results stored in a database. If the test is considered flaky, Guardian notifies the team using Slack and creates an issue in Jira.
```
What do you think? Also, would it be great to have a `Features` subsection in this section?